daemons/execd/remoted_tls.c

```diff
@@ -418,5 +418,10 @@ lrmd_tls_server_destroy(void)
         close(ssock);
         ssock = -1;
     }
+
+    /* If we are in the process of shutting down, then we should actually exit.
+     * bz#1804259
+     */
+    lrmd_exit_if_shutting_down();
```

Comment:
lrmd_tls_server_destroy() is called by lrmd_exit() and lrmd_shutdown() rather than as a connection destroy function which it sounds like

The various connection functions are:
* lrmd_ipc_closed()/lrmd_ipc_destroy(): local client connections on the remote node, I think we can ignore these
* lrmd_remote_client_destroy(): when the cluster's connection to the remote server drops; this calls lrmd_client_destroy(), which should already exit if shutting down, so hopefully this one is ok
* lrmd_remote_connection_destroy(): when the server side connection drops; currently this does nothing, so I'm guessing this is where the problem is (I'm not sure how we could even reproduce this -- it's called by glib when the mainloop source is removed)